source/common/stats/statsd.cc

```diff
@@ -53,24 +54,51 @@ void Writer::send(const std::string& message) {
 }
 
 UdpStatsdSink::UdpStatsdSink(ThreadLocal::SlotAllocator& tls,
-                             Network::Address::InstanceConstSharedPtr address)
-    : tls_(tls.allocateSlot()), server_address_(address) {
+                             Network::Address::InstanceConstSharedPtr address, bool useTag)
+    : tls_(tls.allocateSlot()), server_address_(address), useTag_(useTag) {
   tls_->set([this](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
     return std::make_shared<Writer>(this->server_address_);
   });
 }
 
 void UdpStatsdSink::flushCounter(const Counter& counter, uint64_t delta) {
-  tls_->getTyped<Writer>().writeCounter(counter.name(), delta);
+  tls_->getTyped<Writer>().writeCounter(getName(counter), buildTagStr(counter.tags()), delta);
 }
 
 void UdpStatsdSink::flushGauge(const Gauge& gauge, uint64_t value) {
-  tls_->getTyped<Writer>().writeGauge(gauge.name(), value);
+  tls_->getTyped<Writer>().writeGauge(getName(gauge), buildTagStr(gauge.tags()), value);
 }
 
 void UdpStatsdSink::onHistogramComplete(const Histogram& histogram, uint64_t value) {
   // For statsd histograms are all timers.
-  tls_->getTyped<Writer>().writeTimer(histogram.name(), std::chrono::milliseconds(value));
+  tls_->getTyped<Writer>().writeTimer(getName(histogram), buildTagStr(histogram.tags()),
+                                      std::chrono::milliseconds(value));
+}
+
+std::string UdpStatsdSink::getName(const Metric& metric) {
+  if (useTag_) {
+    return metric.tagExtractedName();
+  } else {
+    return metric.name();
+  }
+}
+
+const std::string buildTagStr(const std::vector<Tag>& tags) {
+  if (tags.empty()) {
```

Comment:
Should this be `use_tag_ || tags.empty()` to preserve existing behavior?

Reply:
Agreed. 80560ce